config/confighttp/confighttp.go

```diff
@@ -85,6 +111,16 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 		}
 	}
 
+	// Compress the body using specified compression methods if non-empty string is provided.
+	// Supporting Gzip, zlib, deflate, snappy, and zstd.
+	if hcs.Compression != "" {
+		if compressionKey := getHTTPCompressionKey(hcs.Compression); compressionKey != CompressionUnsupported {
+			clientTransport = middleware.NewCompressRoundTripper(clientTransport, string(compressionKey))
```

Comment:
If the suggestion about the compression type `none` is accepted, you'll need to handle this special case as well.

Reply:
changed